event-handler/teleport_events_watcher.go

```diff
@@ -130,9 +131,27 @@ func (t *TeleportEventsWatcher) flipPage() bool {
 // fetch fetches the page and sets the position to the event after latest known
 func (t *TeleportEventsWatcher) fetch(ctx context.Context) error {
 	log := logger.Get(ctx)
-
 	b, nextCursor, err := t.getEvents(ctx)
-	if err != nil {
+	// When a trace.BadParameter error is returned, it means that the teleport-plugins
+	// protobuf version is incompatible with the Teleport Auth protobuf version.
+	// This is a fatal error and the plugin should exit because it won't be able to parse the
+	// event that is supported by the newer version of Teleport Auth but not by the
+	// older version of teleport-plugins.
+	// teleport-plugins compatibility is strictly tied to the Teleport Auth version
+	// and the plugin should be updated to the latest version when the Teleport Auth
+	// version is updated. plugins breaks our compatibility promise of supporting
+	// clients 1 major version behind Auth. We don't support older versions of teleport-plugins
+	// with newer versions of Teleport Auth even if they are in the same major version
+	// because we can not guarantee that the plugin will be able to parse the events
+	// introduced between minor or patch versions of Teleport Auth.
+	// This is a temporary solution until we have a better way to handle this.
```

Comment:
It might be worth checking if we do not lose events when Auth Server and any other component are in the same major version but auth is one minor/patch version behind that component  -eg auth in 12.0 but kube agent in 12.1 and introduced a new audit event.
From our compatibility matrix, they are compatible and customers are allowed to use that setup.